File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/FirstAidGuideService.java
package vn.edu.fpt.doghandbook.backend.service;

import org.springframework.web.multipart.MultipartFile;
import vn.edu.fpt.doghandbook.backend.dto.request.FirstAidGuideRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.FirstAidGuideResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;

public interface FirstAidGuideService {

    PageResponse<FirstAidGuideResponse> getAll(int page, int size, String search, String status);

    FirstAidGuideResponse getById(Integer id);

    FirstAidGuideResponse create(FirstAidGuideRequest request, Integer createdByUserId, MultipartFile image);

    FirstAidGuideResponse update(Integer id, FirstAidGuideRequest request, Integer actorUserId, MultipartFile image);

    void delete(Integer id);
}

File: web-admin/src/services/firstAidGuideService.js
import api from './api';

const buildQuery = (page = 0, size = 10, search = '', status = '') => {
  const params = new URLSearchParams();
  params.append('page', String(page));
  params.append('size', String(size));

  if (search?.trim()) params.append('search', search.trim());
  if (status?.trim()) params.append('status', status.trim());

  return params.toString();
};

export const firstAidGuideService = {
  getAll: (page = 0, size = 10, search = '', status = '') =>
    api.get(`/first-aid-guides?${buildQuery(page, size, search, status)}`),
  getById: (id) => api.get(`/first-aid-guides/${id}`),
  create: (data) => api.post('/first-aid-guides', data),
  update: (id, data) => api.put(`/first-aid-guides/${id}`, data),
  delete: (id) => api.delete(`/first-aid-guides/${id}`),
};
